common/types/string.go

```diff
@@ -38,7 +40,14 @@ var (
 		traits.AdderType,
 		traits.ComparerType,
 		traits.MatcherType,
+		traits.ReceiverType,
 		traits.SizerType)
+
+	stringOneArgOverloads = map[string]func(String, ref.Value) ref.Value{
```

Comment:
Question w/o a context: Shouldn't the stringMatches be here?

Reply:
There's a builtin `traits.MatcherType` which other types can implement to support a `matches` function as a first class call. These particular functions are pretty specific to strings though.